hr_attendance_reason/controllers/main.py

```diff
@@ -11,7 +12,23 @@ class HrAttendance(HrAttendance):
     @route("/hr_attendance/attendance_user_data", type="json", auth="user")
     def user_attendance_data(self):
         response = super().user_attendance_data()
-        response.update(self._get_attendance_reason_settings(request.env.company))
+        # try to get the company of the employee to show the correct reasons from
+        # the employees company
+        company = (
+            request.env.user.employee_id.company_id
+            if request.env.user.employee_id
+            else request.env.company
+        )
+        response.update(self._get_attendance_reason_settings(company))
+        # get available reasons for employee company
+        reasons = []
+        if response.get("attendance_state", False):
+            if response.get("attendance_state", False) == "checked_out":
+                action_type = "sign_in"
+            else:
+                action_type = "sign_out"
+            reasons = self._get_attendance_reasons(action_type, company)
+        response.update({"reasons": reasons})
```

Comment:
IMO these changes are not necessary

Reply:
I made the change to get the the reasons of the correct company (company of the employee, see my first comment). Before the reasons were filtered directly in the Javascript code, but it was very difficult to get the correct company at this point. So I decided to source out this part from Javascript to the controller. Another advantage of this is that the logic for the attendance menu and the kiosk is now the same for that. So that's the better way anyway.